elastic4s-client-http4s/src/main/scala/com/sksamuel/elastic4s/http4s/Http4sClient.scala

```diff
@@ -1,87 +1,44 @@
 package com.sksamuel.elastic4s.http4s
 
-import cats.effect.unsafe.IORuntime
-import cats.effect.{Async, IO}
+import cats.effect.Async
 import com.sksamuel.elastic4s
 import com.sksamuel.elastic4s.ElasticNodeEndpoint
 import fs2.io.file.Files
 import org.http4s
 
 import scala.language.higherKinds
 
-// class to support callback interface
-trait CallbackRunner[F[_]] {
-  def run[A](fa: F[A], cb: Either[Throwable, A] => Unit): Unit
-}
-
 object Http4sClient {
 
-  def usingIO(
-      client: http4s.client.Client[IO],
-      endpoint: http4s.Uri
-  )(implicit runtime: IORuntime): Http4sClient[IO] = {
-    val ioRunner = new CallbackRunner[IO] {
-      override def run[A](fa: IO[A], cb: Either[Throwable, A] => Unit): Unit = fa.unsafeRunAsync(cb)
-    }
-
-    new Http4sClient(
-      client = client,
-      endpoint = endpoint,
-      runner = ioRunner
-    )
-  }
-
-  @deprecated("Use usingIO with http4s.Uri", "8.16.0")
-  def usingIO(
-      client: http4s.client.Client[IO],
-      endpoint: ElasticNodeEndpoint
-  )(implicit runtime: IORuntime): Http4sClient[IO] = {
-    val ioRunner = new CallbackRunner[IO] {
-      override def run[A](fa: IO[A], cb: Either[Throwable, A] => Unit): Unit = fa.unsafeRunAsync(cb)
-    }
-
-    Http4sClient(
-      client = client,
-      endpoint = endpoint,
-      runner = ioRunner
-    )
-  }
-
   def apply[F[_]: Async: Files](
```

Comment:
Maybe this is an oversight from last pr, I think we could lower the constraint from `Async` to `Sync` here.

Reply:
Correct, fixed